### PR TITLE
Add oracle provider retry capabilities.

### PR DESCRIPTION
**Pull request by robo-cap**
## Problem Statement

When there are too many secrets to be synced from the Oracle Vault using `oracle` provider, the request may fail with the error: `Service error:TooManyRequests. User-rate limit exceeded.. http status code: 429. Opc`

## Proposed Changes

Add retry support to the `oracle` provider.
It's using the `spec.retrySettings` of the `SecretStore` resource.

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Review by shuheiktgw — COMMENTED** [review]
Hi, @robo-cap 👋 Thanks for your PR! We cannot review nor merge this PR without appropriate unit tests, so would you add them first?

File: pkg/provider/oracle/oracle.go [hunk 1]
```diff
@@ -188,6 +191,41 @@ func (vms *VaultManagementService) NewClient(ctx context.Context, store esv1beta
...
+			retryAmount = 3
+		}
+
+		if storeSpec.RetrySettings.RetryInterval != nil {
+			retryDuration, err = time.ParseDuration(*storeSpec.RetrySettings.RetryInterval)
+			if err != nil {
+				return nil, fmt.Errorf(errRetryInterval, *storeSpec.RetrySettings.RetryInterval)
+			}
+		} else {
+			retryDuration = 5 * time.Second
```
[the remark this answers is not in the document]
Comment (812142) [reply pkg/provider/oracle/oracle.go:211]: pkg/provider/oracle/oracle.go

[the remark this answers is not in the document]
Comment (812142) [reply pkg/provider/oracle/oracle.go:211]: 🤷

[the remark this answers is not in the document]
Comment (robo-cap) [reply pkg/provider/oracle/oracle.go:211]: > Where did these magic default numbers `retryAmount = 3` and `retryDuration = 5` come from? 🤔

I've considered that it makes sense to keep it consistent with the default values used in the retry policy of the other providers.

**Comment by robo-cap** [comment]
> Hi, @robo-cap 👋 Thanks for your PR! We cannot review nor merge this PR without appropriate unit tests, so would you add them first?

Unit tests for the retry policy were added.

**Comment by shuheiktgw** [comment]
Thank you for adding unit tests. I've spotted several minor issues with the change, so instead of pointing them out and having you fix them, I decided to take it over to save some time. Please feel free to leave your comments on the PR!

https://github.com/external-secrets/external-secrets/pull/2762

**Pull request closed without merging**